main.go

```diff
@@ -19,11 +19,11 @@ func Run(args []string) error {
 	if err != nil {
 		return err
 	}
-	err = LaunchProcess(cfg, args, os.Stdout, os.Stderr)
+	doUpgrade, err := LaunchProcess(cfg, args, os.Stdout, os.Stderr)
 
 	// if RestartAfterUpgrade, we launch after a successful upgrade (only condition LaunchProcess returns nil)
-	for cfg.RestartAfterUpgrade && err == nil {
-		err = LaunchProcess(cfg, args, os.Stdout, os.Stderr)
+	for cfg.RestartAfterUpgrade && err == nil && doUpgrade {
+		_, err = LaunchProcess(cfg, args, os.Stdout, os.Stderr)
```

Comment:
since this is a loop, we should have `doUpgrade, err = ` here as well, right?

(in the given use case, this probably doesn't matter, but good for correctness in unexpected cases)